### PR TITLE
Clamp noise in `FixedGaussianNoise` via `min_fixed_noise` context manager

### PR DESCRIPTION
**Pull request by saitcakmak**
Introduces a `min_fixed_noise` context manager that is used to clamp the `noise` values in `FixedGaussianNoise` to a minimum value dependent on the `dtype`. This helps eliminate some numerical issues we face in downstream when using too small (0) noise. 

Test plan: Units.

cc @Balandat, @mpolson64, @gpleiss

### CODE REVIEW

File: gpytorch/settings.py [hunk 1]
```diff
@@ -420,6 +420,21 @@ class max_cg_iterations(_value_context):
...
 
 
+class min_fixed_noise(_dtype_value_context):
+    """
+    The minimum noise value that can be used in :obj:`~gpytorch.likelihoods.FixedNoiseGaussianLikelihood`.
+    If the supplied noise values are smaller than this, they are rounded up and a warning is raised.
+
+    - Default for `float`: 1e-6
+    - Default for `double`: 1e-10
+    - Default for `half`: 1e-3
```
Comment (Balandat) [inline gpytorch/settings.py:430]: this doesn't seem to agree with the values below?

Comment (Balandat) [reply gpytorch/settings.py:430]: also 1e-8 is pretty small... seems like this wouldn't resolve the issues we've seen with numerical issues creeping up in the backward pass?

Comment (saitcakmak) [reply gpytorch/settings.py]: 1e-8 seemed to work for our particular case, but swapped to 1e-6 / 1e-4 for more conservative values.

File: test/likelihoods/test_gaussian_likelihood.py [hunk 2]
```diff
@@ -84,6 +85,13 @@ def test_fixed_noise_gaussian_likelihood(self, cuda=False):
             obs_noise = 0.1 + torch.rand(5, device=device, dtype=dtype)
             out = lkhd(mvn, noise=obs_noise)
             self.assertTrue(torch.allclose(out.variance, 1 + obs_noise))
+            # test noise smaller than min_fixed_noise
+            expected_min_noise = settings.min_fixed_noise.value(dtype)
+            noise[:2] = 0
+            lkhd = FixedNoiseGaussianLikelihood(noise=noise)
```
Comment (Balandat) [inline test/likelihoods/test_gaussian_likelihood.py:91]: Should we also check that the warning is raised here? Not super critical though.

**Review by Balandat — APPROVED** [review]

**Pull request merged**